cmd/erasure-healing.go

```diff
@@ -265,8 +265,12 @@ func (er erasureObjects) healObject(ctx context.Context, bucket string, object s
 			// If data is sane on any one disk, we can
 			// extract the correct object size.
 			result.ObjectSize = partsMetadata[i].Size
-			result.ParityBlocks = partsMetadata[i].Erasure.ParityBlocks
-			result.DataBlocks = partsMetadata[i].Erasure.DataBlocks
+			if partsMetadata[i].Erasure.ParityBlocks != 0 {
+				result.ParityBlocks = partsMetadata[i].Erasure.ParityBlocks
+			}
+			if partsMetadata[i].Erasure.DataBlocks != 0 {
```

Comment:
```suggestion
			if partsMetadata[i].Erasure.DataBlocks > 0 {
```

Reply:
combine the two if-checks together as `&&`

Reply:
if DataBlocks is empty we should set the value to `len(storageDisks)/2`

Reply:
> combine the two if-checks together as `&&`
Done


> if DataBlocks is empty we should set the value to `len(storageDisks)/2`
This is already done in the beginning of the function